tests/TestCase/Routing/RouterTest.php

```diff
@@ -2753,6 +2767,55 @@ public function testReverseWithExtension()
         $this->assertEquals($expected, $result);
     }
 
+    public function testReverseToArrayQuery()
+    {
+        Router::connect('/:lang/:controller/:action/*', [], ['lang' => '[a-z]{3}']);
+        $params = [
+            'lang' => 'eng',
+            'controller' => 'posts',
+            'action' => 'view',
+            'pass' => [123],
+            'url' => ['url' => 'eng/posts/view/123', 'foo' => 'bar', 'baz' => 'quu'],
+            'paging' => [],
+            'models' => [],
```

Comment:
This test is better. It better captures which keys are omitted and which are preserved.